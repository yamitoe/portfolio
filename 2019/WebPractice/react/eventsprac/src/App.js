import React from 'react';
import './stylesheets/App.scss';
import { AddItem } from './components/AddItem.js';

export function App() {
  return <AddItem />;
}
