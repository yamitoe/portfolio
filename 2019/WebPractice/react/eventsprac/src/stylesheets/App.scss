$contwidth: 180px;
.container-items {
  display: flex;
  flex-wrap: wrap;
  .container-item {
    margin: 5px;
    img {
      display: block;
      width: $contwidth;
      height: $contwidth;
      background-color: black;
    }
    div.desc,
    h2 {
      overflow-wrap: break-word;
      width: $contwidth;
      margin: 0px;
    }
  }
}

.container-addItem {
  border: 1px solid;
  width: $contwidth;
  button {
    width: 100%;
  }
  img {
    margin: 5px;
    width: 40px;
  }
}
#handleSubmit {
  background-color: gray;
  position: fixed;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  background-color: hsla(4, 20%, 10%, 0.4);
  justify-content: center;
  align-items: center;

  form {
    display: flex;
    flex-direction: column;
    width: 70%;
    height: 70%;
    padding: 20px;
    background-color: hsla(6, 45%, 88%, 1);
    border: 1px solid black;
    overflow: auto;
    max-width: 450px;
    max-height: 400px;
    input {
      width: 100%;
      margin: 5px 0 5px 0;
    }
    .textarea {
      font-family: Arial, Helvetica, sans-serif;
      font-size: 12px;
      flex: 1 0 50px;
      max-height: 60px;
      //For firefox specific problem
      &:invalid {
        box-shadow: none;
      }
    }
  }
}
