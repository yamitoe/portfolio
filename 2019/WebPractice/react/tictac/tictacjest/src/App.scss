.board{
    div{
        display: flex;
        span{
            display: block;
            flex-basis: 35px;
            height: 30px;
            padding: 10px;
            border: 1px solid black;
            text-align: center;
        }
    }

}