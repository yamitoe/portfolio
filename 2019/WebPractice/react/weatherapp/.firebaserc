{
  "projects": {
    "default": "weatherapp-5c23d"
  }
}
