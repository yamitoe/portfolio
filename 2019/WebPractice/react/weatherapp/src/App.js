import React from "react";
import { route } from "./components/shared/routing";
import { Link } from "react-router-dom";

export default class App extends React.Component {
  render() {
    return (
      <React.Fragment>
        <Link to="./" id="home">
          Home
        </Link>
        {route}
      </React.Fragment>
    );
  }
}
