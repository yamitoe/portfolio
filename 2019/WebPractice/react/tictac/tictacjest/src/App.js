import React, {useState} from 'react';
import './App.scss';
import {Square} from './Square';

function Board(){
  const arraryInit = 9;
  const [turn, setTurn] = useState("X");
  const [gameState,setGameState] =useState({
    board:Array(arraryInit).fill(null),
    history: [Array(arraryInit).fill(null)]
  });
  const [counter, setCounter] = useState(0);

  let handleClick = (index)=>{
    if(gameState.board[index] === null){
      handleBoard(index);
      handleTurn();
    }
  }
  let handleBoard = (index) =>{
    setGameState(prevBoard=>{
      let currBoard = prevBoard.board.slice();
      let currHistory;
      currBoard.splice(index,1,turn);
      //Means a undo has been done
      if(counter > 0){
        let currIndex = (gameState.history.length -1) - (counter);
        currHistory = prevBoard.history.slice(0,currIndex+1);
        setCounter(0);
      }
      else{
        currHistory = prevBoard.history.slice();
      }
      currHistory.push(currBoard); 
      return {board: currBoard, history: currHistory};
    });
  }
  
  let handleTurn =()=>{
    setTurn(prevTurn=>(prevTurn === "X") ? "O" : "X");
  }

  let buttonUndo = ()=>{
    let currIndex = (gameState.history.length -1) - (counter + 1);
    //Case when index is at 0
    if(currIndex > 0){
      setCounter(i=>i+1);
    }
    //change the currently displayed board
    setGameState(prevBoard =>{
      //copy data
      let arr = prevBoard.history.slice();
      //Get the previous board from history
      let [lastArr] = arr.splice(currIndex,1);
      let merge = Object.assign(prevBoard,{board:lastArr})
      return merge;
    });
    
  }

  let buttonRedo = ()=>{
    let currIndex = (gameState.history.length -1) - (counter - 1);

    if(currIndex < gameState.history.length){
      setCounter(i=>i-1);
    }
    //change the currently displayed board
    setGameState(prevBoard =>{
      //copy data
      let arr = prevBoard.history.slice();
      //Get the previous board from history
      let [lastArr] = arr.splice(currIndex,1);
      let merge = Object.assign(prevBoard,{board:lastArr})
      return merge;
    });
  }

  let makeSquare = (box)=>{
    return <Square num={gameState.board[box]} handleClick={()=> handleClick(box)}/>
  }



  return(
    <section className="board">
      <div>
        {makeSquare(0)}
        {makeSquare(1)}
        {makeSquare(2)}
      </div>
      <div>
        {makeSquare(3)}
        {makeSquare(4)}
        {makeSquare(5)}
      </div>
      <div>
        {makeSquare(6)}
        {makeSquare(7)}
        {makeSquare(8)}
      </div>

      <button onClick={buttonUndo}>Undo</button>
      <button onClick={buttonRedo}>Redo</button>
    </section>
  )
    
}


function App() {
  return (
    <div className="App">
      <Board/>
    </div>
  );
}

export default App;
