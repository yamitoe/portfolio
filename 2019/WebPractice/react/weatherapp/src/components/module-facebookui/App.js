import React, { useState, useEffect } from "react";
import "./App.scss";
import { NavBar } from "./components/NavBar";
import { NavItem } from "./components/NavItem";
import { DropDownMenu } from "./components/DropDownMenu";
import { SampleData } from "./components/SampleData";
//SVG
import { ReactComponent as Arrow } from "./icons/arrow.svg";
import { ReactComponent as Plus } from "./icons/plus.svg";
import { ReactComponent as Bolt } from "./icons/bolt.svg";
import { ReactComponent as Bell } from "./icons/bell.svg";
import { ReactComponent as Caret } from "./icons/caret.svg";

export function App() {
  //For theme changer
  const [checked, setChecked] = useState(true);

  //on first load theme is set to dark
  useEffect(() => {
    //if empty, dark theme is default
    if (localStorage.getItem("theme") === null) {
      localStorage.setItem("theme", "dark");
    }
    document.body.setAttribute("theme", localStorage.getItem("theme"));
    setChecked(localStorage.getItem("theme") === "dark" ? true : false);
  }, []);

  return (
    <div className="App">
      {document.body.removeAttribute("backColor")}
      <NavBar>
        <NavItem img={<Arrow />} />
        <NavItem img={<Plus />} />
        <NavItem img={<Bolt />} />
        <NavItem img={<Caret />}>
          <DropDownMenu checked={checked} checkedFunc={setChecked} />
        </NavItem>
      </NavBar>
      <SampleData />
    </div>
  );
}

export default App;
