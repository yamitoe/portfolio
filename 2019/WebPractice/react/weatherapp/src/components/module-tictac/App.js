import React, { useState } from "react";
import "./App.scss";
import { Square } from "./Square";

function Board() {
  const arraryInit = 9;
  const [turn, setTurn] = useState("X");
  const [gameState, setGameState] = useState({
    board: Array(arraryInit).fill(null),
    history: [Array(arraryInit).fill(null)]
  });
  const [counter, setCounter] = useState(0);
  const [winner, setWinner] = useState(false);

  let handleClick = index => {
    if (gameState.board[index] === null) {
      handleBoard(index);
      handleTurn();
    }
  };
  // Update game history //Update board //Reset Counter
  let handleBoard = index => {
    setGameState(prevBoard => {
      let currBoard = prevBoard.board.slice();
      let currHistory;
      currBoard.splice(index, 1, turn);
      //Means a undo has been done
      if (counter > 0) {
        let currIndex = gameState.history.length - 1 - counter;
        currHistory = prevBoard.history.slice(0, currIndex + 1);
        setCounter(0);
      } else {
        currHistory = prevBoard.history.slice();
      }
      winCondition(currBoard);
      currHistory.push(currBoard);
      return { board: currBoard, history: currHistory };
    });
  };

  let handleTurn = () => {
    setTurn(prevTurn => (prevTurn === "X" ? "O" : "X"));
  };
  //Game winning logic
  let winCondition = currBoard => {
    let size = 3;
    //Make the horizontal win condition
    for (let x = 0; x < currBoard.length; x += 3) {
      if (x % size === 0) {
        let rowWin = currBoard.slice(x, x + size);
        checkWinnerFunc(rowWin, "X");
        checkWinnerFunc(rowWin, "O");
      }
    }
    //Vert test
    for (let x = 0; x < size; x++) {
      let vertWin = [
        currBoard[x],
        currBoard[x + size],
        currBoard[x + size + size]
      ];
      checkWinnerFunc(vertWin, "X");
      checkWinnerFunc(vertWin, "O");
    }
    //Diagonal
    for (let x = 0; x < size - 1; x++) {
      let diagonalWin = [
        currBoard[x],
        currBoard[x + size + 1],
        currBoard[x + size + size + 2]
      ];
      checkWinnerFunc(diagonalWin, "X");
      checkWinnerFunc(diagonalWin, "O");
    }
  };

  let checkWinnerFunc = (test, player) => {
    //If someone won
    if (test.every(match => match === player) === true) {
      //If no one has won yet
      if (winner === false) {
        setWinner(player + " Wins");
        //Set board to finished
        setGameState(prevData => {
          let curBoard = prevData.board.slice();
          let curHistory = prevData.history.slice();
          for (let x = 0; x < gameState.board.length; x++) {
            if (curBoard[x] === null) {
              curBoard[x] = true;
            }
          }
          curHistory.push(curBoard);
          return { ...prevData, board: curBoard };
        });
      }
    }
  };
  //Hooks doesn't shallow merge so if using object form must do this
  let buttonUndo = () => {
    //Minus by 1 more value
    let currIndex = gameState.history.length - 1 - counter;
    let maxCount = gameState.history.length - 1; //Every jump in the list

    let toIndex =
      counter < maxCount
        ? gameState.history.length - 1 - (counter + 1)
        : currIndex;
    //Case when index is 0 and above //If not outofbounds

    if (currIndex > 0) {
      handleTurn();
      setWinner(false);
      setCounter(i => i + 1);
    }
    //change the currently displayed board
    buttonMergeState(toIndex);
  };

  let buttonRedo = () => {
    let currIndex = gameState.history.length - 1 - counter;
    //IF only one index return current index //Minus by one less value (going up)
    let toIndex =
      counter > 0 ? gameState.history.length - 1 - (counter - 1) : currIndex;

    if (currIndex < gameState.history.length - 1) {
      handleTurn();
      setCounter(i => i - 1);
    }
    buttonMergeState(toIndex, "redo");
  };

  let buttonMergeState = (toIndex, test) => {
    //change the currently displayed board
    setGameState(prevBoard => {
      let arr = prevBoard.history.slice();
      //Get the previous board from history
      let [lastArr] = arr.splice(toIndex, 1);
      let merge = { ...prevBoard, board: lastArr };
      if (test === "redo") {
        winCondition(lastArr);
      }
      return merge;
    });
  };

  let makeSquare = box => {
    return (
      <Square num={gameState.board[box]} handleClick={() => handleClick(box)} />
    );
  };

  return (
    <section className="board">
      <div>
        {makeSquare(0)}
        {makeSquare(1)}
        {makeSquare(2)}
      </div>
      <div>
        {makeSquare(3)}
        {makeSquare(4)}
        {makeSquare(5)}
      </div>
      <div>
        {makeSquare(6)}
        {makeSquare(7)}
        {makeSquare(8)}
      </div>
      <div>
        <button onClick={buttonUndo}>Undo</button>
        <button onClick={buttonRedo}>Redo</button>
      </div>
      {winner ? <h2>{winner}</h2> : false}
    </section>
  );
}

function App() {
  return (
    <div className="App">
      {document.body.setAttribute("backColor", "tictac")}
      <Board />
    </div>
  );
}

export default App;
