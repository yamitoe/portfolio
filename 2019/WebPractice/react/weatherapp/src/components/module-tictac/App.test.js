import React from "react";
import { render, fireEvent, findAllByTitle } from "@testing-library/react";
import App from "./App";

it("renders", () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

it("has Undo", () => {
  const { getByText } = render(<App />);
  const undoBtn = getByText("Undo");
});

it("has Redo", () => {
  const { getByText } = render(<App />);
  const undoBtn = getByText("Undo");
});

it("fires button", () => {
  const { getByText } = render(<App />);
  fireEvent.click(getByText("Undo"));
});
