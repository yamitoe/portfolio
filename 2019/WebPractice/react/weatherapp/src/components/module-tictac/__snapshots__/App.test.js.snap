// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <section
      class="board"
    >
      <div>
        <span />
        <span />
        <span />
      </div>
      <div>
        <span />
        <span />
        <span />
      </div>
      <div>
        <span />
        <span />
        <span />
      </div>
      <div>
        <button>
          Undo
        </button>
        <button>
          Redo
        </button>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
